bundles/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/internal/ExtendedDeviceStateUtil.java

```diff
@@ -46,4 +52,15 @@ public static String bytesToHex(byte[] bytes) {
     public static byte[] stringToBytes(String input) {
         return input.getBytes(StandardCharsets.ISO_8859_1);
     }
+
+    /**
+     * Convert string to Number:Temperature state with unit Celcius
+     */
+    public static State getTemperatureState(String s) throws NumberFormatException {
+        if (s.equals(TEMPERATURE_UNDEFINED)) {
```

Comment:
Can you please reverse ? You should certainly have a SAT warning on this line.

Reply:
Can you elaborate, I don't understand?

Reply:
Constant at left: TEMPERATURE_UNDEFINED.equals(s)

Maybe the SAT check is more clever than I imagine and only alerts when the variable can be null ?

Reply:
go to the bindings \target\code-analysis folder after you compile and read the report for warnings.

Yes I believe it is more clever as I made a suggestion to someone else and got told it did not apply.

Reply:
Thanks, fixed.